verl/single_controller/base/decorator.py

```diff
@@ -92,42 +92,28 @@ def _split_args_kwargs_data_proto(chunks, *args, **kwargs):
 def _split_args_kwargs_data_proto_with_auto_padding(chunks, *args, **kwargs):
     from verl.protocol import DataProto, DataProtoFuture
 
-    splitted_args = []
-    splitted_kwargs = {}
-
     data_proto_len = None
     padding_size = None
-    for arg in args:
-        assert isinstance(arg, DataProto | DataProtoFuture)
-        if isinstance(arg, DataProto) and arg.is_padding_enabled():
+
+    def _padding_and_split_data(obj, chunks):
+        nonlocal data_proto_len, padding_size
+        assert isinstance(obj, DataProto | DataProtoFuture)
+        if isinstance(obj, DataProto) and obj.is_padding_enabled():
             # for padding, we only support DataProto with same length
             if data_proto_len is None:
-                data_proto_len = len(arg)
+                data_proto_len = len(obj)
                 padding_size = (chunks - (data_proto_len % chunks)) if (data_proto_len % chunks > 0) else 0
-                splitted_kwargs[_padding_size_key] = padding_size
             else:
-                assert data_proto_len == len(arg), (
-                    f"expecting all arg share same length of {data_proto_len}, but got {len(arg)}"
+                assert data_proto_len == len(obj), (
+                    f"expecting all arg share same length of {data_proto_len}, but got {len(obj)}"
                 )
-                data_proto_len = len(arg)
-            arg.padding(padding_size=padding_size)
-
-        splitted_args.append(arg.chunk(chunks=chunks))
+            obj.padding(padding_size=padding_size)
+        return obj.chunk(chunks=chunks)
 
-    for key, val in kwargs.items():
-        assert isinstance(val, DataProto | DataProtoFuture)
-        if isinstance(val, DataProto) and val.is_padding_enabled():
-            # for padding, we only support DataProto with same length
-            if data_proto_len is None:
-                data_proto_len = len(val)
-                padding_size = chunks - (data_proto_len % chunks)
-                splitted_kwargs[_padding_size_key] = padding_size
-            else:
-                assert data_proto_len == len(val), (
-                    f"expecting all arg share same length of {data_proto_len}, but got {len(val)}"
-                )
-                data_proto_len = len(val)
-        splitted_kwargs[key] = val.chunk(chunks=chunks)
+    splitted_args = [_padding_and_split_data(arg, chunks) for arg in args]
+    splitted_kwargs = {key: _padding_and_split_data(val, chunks) for key, val in kwargs.items()}
```

Comment:
Could you add or fix unit test in `test_auto_padding_on_cpu.py`? I think we may lack test case covering the kwargs.